src/Engine/ProtoCore/SyntaxAnalysis/AstVisitor.cs

```diff
@@ -0,0 +1,400 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using ProtoCore.AST.AssociativeAST;
+using ProtoCore.AST;
+
+namespace ProtoCore.SyntaxAnalysis
+{
+    public class AssociativeAstVisitor
+    {
+        public virtual void DefaultVisit(AssociativeNode node)
+        {
+        }
+
+        public virtual void Visit(Node node)
+        {
+            AssociativeNode assocNode = node as AssociativeNode;
+            if (assocNode != null)
+                assocNode.Accept(this);
+        }
+
+        public virtual void VisitCommentNode(CommentNode node)
```

Comment:
Ill need to brush up on visitor patterns, but does this mean that each ASTNode must implement its own DefaultVisit? or Visit function?